### PR TITLE
Add CUDA 12.0.1 support

### PR DESCRIPTION
**Pull request by raydouglass**
Adds support for CUDA 12.0

Depends on #562

### CODE REVIEW

File: matrix-test.yaml [hunk 1]
```diff
@@ -1,9 +1,11 @@
 pull-request:
   - { CUDA_VER: '11.2.2', ARCH: 'amd64', PYTHON_VER: '3.9', GPU: 'v100', DRIVER: 'earliest' }
   - { CUDA_VER: '11.8.0', ARCH: 'amd64', PYTHON_VER: '3.10', GPU: 'v100', DRIVER: 'latest' }
-  - { CUDA_VER: '11.8.0', ARCH: 'arm64', PYTHON_VER: '3.10', GPU: 'a100', DRIVER: 'latest' }
+  - { CUDA_VER: '12.0.1', ARCH: 'arm64', PYTHON_VER: '3.10', GPU: 'a100', DRIVER: 'latest' }
```
Comment (bdice) [inline matrix-test.yaml:4]: We don't support CUDA 12 conda packages on ARM yet.

File: matrix-test.yaml [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 nightly:
   - { CUDA_VER: '11.2.2', ARCH: 'amd64', PYTHON_VER: '3.9', GPU: 'v100', DRIVER: 'earliest' }
   - { CUDA_VER: '11.2.2', ARCH: 'amd64', PYTHON_VER: '3.9', GPU: 'v100', DRIVER: 'latest' }
   - { CUDA_VER: '11.8.0', ARCH: 'amd64', PYTHON_VER: '3.10', GPU: 'v100', DRIVER: 'latest' }
   - { CUDA_VER: '11.8.0', ARCH: 'arm64', PYTHON_VER: '3.10', GPU: 'a100', DRIVER: 'latest' }
+  - { CUDA_VER: '12.0.1', ARCH: 'amd64', PYTHON_VER: '3.10', GPU: 'v100', DRIVER: 'latest' }
+  - { CUDA_VER: '12.0.1', ARCH: 'arm64', PYTHON_VER: '3.10', GPU: 'a100', DRIVER: 'latest' }
```
Comment (bdice) [inline matrix-test.yaml:11]:
```suggestion
```

File: matrix-test.yaml (hunk 1 above)
Comment (raydouglass) [reply matrix-test.yaml:4]: Whoops, I forgot about this.

**Comment by raydouglass** [comment]
I see there is a failure now because there are no CUDA 12 test notebook dependencies for `cusignal`. I'm discussing which notebooks to remove from the tests later today and will likely be removing `cusignal`, so I'll update this PR after that.

**Review by bdice — APPROVED** [review]
A couple CI jobs are still running but this looks good to me.

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**